### ISSUE TITLE
Problem with secp256r1 curve ASN1 identifier

### ISSUE THREAD

**Issue by reubent** [kjur/jsrsasign#333]
```
const csri = new KJUR.asn1.csr.CertificationRequestInfo();
csri.setSubjectPublicKeyByGetKey(...keydata...);
const csr = new KJUR.asn1.csr.CertificationRequest({'csrinfo': csri});
csr.sign("SHA256withRSA", ...privatekeydata...);
```
throws `algId and/or subjPubKey not set`

If I boil it down a bit
```
const key = KEYUTIL.getKey("... keydata ...")
let asn1Params = KJUR.asn1.DERObjectIdentifier({name: key.curveName})
```
I get `Uncaught DERObjectIdentifier oidName undefined: P-256`

Looking at the code, it's doing
```
var asn1Params = new _DERObjectIdentifier({'name': key.curveName});
```
on line 1857 of `asn1x509-1.0.js`  which returns `P-256` whereas the mapping for the identifier in `asn1x509-1.0.js` is `secp256r1` (line 2475)

I'm happy to submit a patch but I'm not sure which end you'd prefer changed - the ASN mapping or the algorithm setting

**Comment by MaxPhilips** [comment]
hey @kjur - I've run into this issue as well. any preference on how this should be addressed as mentioned above? i'm also happy to submit a patch

**Comment by kjur** [comment]
Hi @MaxPhilips , thank you for your response. Could you provide a patch?

**Comment by MaxPhilips** [comment]
@kjur I opened this PR https://github.com/kjur/jsrsasign/pull/461

**Comment by kjur** [comment]
This issue was fixed in the release 10.0.5 today. Thank you @MaxPhilips .
